01-syntax/03-data-structures/deep-comparation/deepEqual.js

```diff
@@ -4,15 +4,49 @@
  */
 /* global deepEqual */
 
-// → Your code here.
-
 var obj = {
   here: {
     is: 'an'
   },
   object: 2
 };
 
+function deepEqual(firstObj, secondObj) {
+  var firstKey;
+  var secondKey;
+  var same;
+
+  arguments[0] = firstObj || {};
+  arguments[1] = secondObj || {};
+
+  if (Object.keys(firstObj).length !== Object.keys(secondObj).length) {
+    return false;
+  } else if (firstObj === secondObj) {
+    return true;
+  }
+
+  for (firstKey in firstObj) {
+    if (firstObj.hasOwnProperty(firstKey)) {
+      same = false;
+      for (secondKey in secondObj) {
+        if (secondObj.hasOwnProperty(secondKey) &&
```

Comment:
Looks quite complex for me,  could you extract 'isObject' functionality to `isObject` function
It may make your code more readable. 

> keep `isObject` function in `deepEqual`